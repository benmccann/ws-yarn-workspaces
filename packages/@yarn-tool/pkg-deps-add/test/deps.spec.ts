//@noUnusedParameters:false

import { basename, extname } from 'path';
import { IPackageJson } from '@ts-type/package-dts/package-json';
import { addDependencies, addDependenciesIfNotExists } from '../index';

beforeAll(async () =>
{

});

describe(basename(__filename, extname(__filename)), () =>
{
	let pkg: IPackageJson;

	beforeAll(async () => {
		pkg = {
			dependencies: {
				"tslib": "^2"
			}
		}
	});

	test(`addDependenciesIfNotExists`, () =>
	{

		let actual = addDependenciesIfNotExists(pkg, 'tslib2', '^2.3.1');

		expect(pkg).toHaveProperty('dependencies.tslib2', '^2.3.1');
		expect(actual).toMatchSnapshot();

		addDependencies(pkg, 'tslib', '^2.3.1', {});

		expect(pkg).toHaveProperty('dependencies.tslib', '^2');

		addDependencies(pkg, 'tslib', '^2.3.1', {}, true);

		expect(pkg).toHaveProperty('dependencies.tslib', '^2.3.1');

	});

})
