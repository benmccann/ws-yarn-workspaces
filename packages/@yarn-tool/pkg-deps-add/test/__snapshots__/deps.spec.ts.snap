// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deps.spec addDependenciesIfNotExists 1`] = `
Object {
  "bool": 2,
  "pkg": Object {
    "dependencies": Object {
      "tslib": "^2",
      "tslib2": "^2.3.1",
    },
  },
}
`;
